JustSaying/Messaging/MessageProcessingStrategies/ThrottledOptions.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using JustSaying.Messaging.Monitoring;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Messaging.MessageProcessingStrategies
+{
+    /// <summary>
+    /// A class representing options for the <see cref="Throttled"/> class.
+    /// </summary>
+    public class ThrottledOptions
+    {
+        /// <summary>
+        /// Gets or sets the maximum number of messages to process concurrently.
+        /// </summary>
+        public int MaxConcurrency { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to process messages by dispatching
+        /// them to the thread pool or whether to process messages before returning.
+        /// </summary>
+        public bool UseThreadPool { get; set; }
```

Comment:
How about:

```csharp
public ProcessingMode ProcessingMode { get; set; }

enum  ProcessingMode
{
    Sequential,
    Parallel
}
```
 to make it clear how this changes the behaviour?

Reply:
Actually that’s nicer as it’s more explicit. Will change.

Reply:
Although...that does mean an enum specific to a specific options class of a specific implementation of an interface... 🤔

Reply:
Compromised by renaming the `bool` and inverting it to mean `true` to do sequential processing (the new opt-in behaviour): fef2976c26a24c1b9c1b26a47fc02e29728a1a6e

Reply:
Yeah, I went through the same thought process, I think it's worth it here. A rename would also suffice, I thought that `UseThreadPool` might not convey the behaviour (assuming people don't read the doc comments).

Reply:
I think I prefer the rename as we're not adding a whole enum with only two possible values that then we'd have to error handle any other possible value someone could fudge in.